node_registry.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"github.com/microsoft/BladeMonRT/nodes"
+)
+
+func makeInstance(typeName string) nodes.InterfaceNode {
```

Comment:
Let's make this into a class and call it "NodeFactory"

Let `NodeFactory` have a public function `ConstructNode` given the string name of the node (basically just a copy of makeInstance here

Reply:
The WorkflowFactory can call this function

Reply:
Makes sense. I'm hesistant to create a class for this because this will be an empty struct NodeFactory. I changed the file name to node factory.

Reply:
I think there's no harm in creating an empty struct. Done.